### PR TITLE
update nixos ucm install

### PR DESCRIPTION
**Pull request by dj1ch**
pretty much the title, just updated the commit hash, and some info on nixos config locations. install script wise nothing seemed to be out of the ordinary, copied the ucm to `/usr/share/alsa` like normal. didn't want the nixos page to become outdated @WeirdTreeThing

### CODE REVIEW

File: src/docs/installing/distros.md [hunk 1]
```diff
@@ -145,4 +153,4 @@ boot.extraModprobeConfig = ''
 boot.extraModprobeConfig = ''
   options snd-intel-dspcfg dsp_driver=4
 '';
-```
+```
```
Comment (ethanaobrien) [inline src/docs/installing/distros.md:156]: No newline?

**Review by ethanaobrien — CHANGES_REQUESTED** [review]

File: src/docs/installing/distros.md (hunk 1 above)
Comment (dj1ch) [reply src/docs/installing/distros.md:156]: originally i planned on putting a tip down there, i moved it back to the top.

**Comment by ChocolateLoverRaj** [comment]
@dj1ch did you test the nixos instructions on a Chromebook? If you did, can you provide the full nix config? I'm trying the nixos guide and it's compiling stuff for over 10 minutes and taking up a lot of storage space. I think the nixos guide needs some fixing in addition to just updating the commit used.

File: src/docs/installing/distros.md [hunk 2]
```diff
@@ -89,7 +97,7 @@ If your generation isn't listed above, you can skip this section
       wttsrc = (fetchFromGitHub {
         owner = "WeirdTreeThing";
         repo = "chromebook-ucm-conf";
-        rev = "484f5c581ac45c4ee6cfaf62bdecedfa44353424";
+        rev = "6d8835f89c7c7e5fac7b6aaea0e36a6b1d1e4be5";
         hash = "sha256-Jal+VfxrPSAPg9ZR+e3QCy4jgSWT4sSShxICKTGJvAI=";
```
Comment (ninelore) [inline src/docs/installing/distros.md:101]: changed commit but not the checksum?

**Review by ninelore — CHANGES_REQUESTED** [review]

File: src/docs/installing/distros.md (hunk 2 above)
Comment (ChocolateLoverRaj) [reply src/docs/installing/distros.md:101]: Doubt

Comment (ChocolateLoverRaj) [reply src/docs/installing/distros.md:101]: If `rev` is changed then `hash` should've also been changed.

**Comment by ChocolateLoverRaj** [comment]
@dj1ch what about the issue of ["50GB of rebuilds"](https://discourse.nixos.org/t/help-with-avoiding-50gb-worth-of-rebuils/41655)?

**Comment by dj1ch** [comment]
> @dj1ch what about the issue of ["50GB of rebuilds"](https://discourse.nixos.org/t/help-with-avoiding-50gb-worth-of-rebuils/41655)?

ok, pretty much did the same thing you did with audio.nix, tested it on my machine and it runs fine.

**Comment by ChocolateLoverRaj** [comment]
A few issues:
- The ucm-conf is a few months old (Should work for most devices, but ideally it should use the latest ucm-conf which will also work for all generations. I'm planning on changing the audio config to use the latest version, but let me know if you want to do it and then I won't, and I'll just test it out once you do.
- There appear to be instructions for AVS but does AVS work? Do some files need to be added for AVS?
- What about max98357a (the danger)?
- AMD audio not implemented yet (this is ok but maybe it should be mentioned in the docs)
- Not needed, but `51-increase-headroom.lua` should also be added

**Comment by ChocolateLoverRaj** [comment]
THis is my config btw: https://github.com/ChocolateLoverRaj/nixos-system-config/blob/84c9e4034a2fb85b08ac4b3a7487ba163efd860b/audio.nix

**Comment by dj1ch** [comment]
> A few issues:
> 
> * The ucm-conf is a few months old (Should work for most devices, but ideally it should use the latest ucm-conf which will also work for all generations. I'm planning on changing the audio config to use the latest version, but let me know if you want to do it and then I won't, and I'll just test it out once you do.
> * There appear to be instructions for AVS but does AVS work? Do some files need to be added for AVS?
> * What about max98357a (the danger)?
> * AMD audio not implemented yet (this is ok but maybe it should be mentioned in the docs)
> * Not needed, but `51-increase-headroom.lua` should also be added
> * And you can use fetch from GitHub instead of fetching a zip

we can probably try using the latest commit if possible. ill test it and ill let you know.

**Comment by WeirdTreeThing** [comment]
bruh why are you still using that repo

**Comment by ChocolateLoverRaj** [comment]
> bruh why are you still using that repo

Should we use https://github.com/WeirdTreeThing/alsa-ucm-conf-cros/tree/standalone ? Is that "your upstream" repo for Chromebook audio, and not chromebook-ucm-conf?

**Comment by dj1ch** [comment]
> > bruh why are you still using that repo
> 
> Should we use https://github.com/WeirdTreeThing/alsa-ucm-conf-cros/tree/standalone ? Is that "your upstream" repo for Chromebook audio, and not chromebook-ucm-conf?

probably this [issue](https://github.com/batocera-linux/batocera.linux/issues/11006#issuecomment-1935830865) sums it up

**Comment by ChocolateLoverRaj** [comment]
Except that we could just use the latest commit of `chromebook-ucm-conf` and it will work for all SOF generations. So I'm still confused.

**Comment by ChocolateLoverRaj** [comment]
I mean https://ryantm.github.io/nixpkgs/builders/fetchers/#fetchfromgithub

**Comment by ChocolateLoverRaj** [comment]
Also include `51-increase-headroom.lua`. That's what I did in [my config](https://github.com/ChocolateLoverRaj/nixos-system-config/blob/84c9e4034a2fb85b08ac4b3a7487ba163efd860b/audio.nix). There are different instructions for `23.11` and `unstable`.

**Comment by WeirdTreeThing** [comment]
what about the avs dmic config?

**Comment by ninelore** [comment]
@ChocolateLoverRaj Mind if you could test this? We currently have no nix users in the Contrib team

**Review by ninelore — CHANGES_REQUESTED** [review]
Why dont we suggest to put both keyd and audio stuff in a common chrome-device.nix and import that into the main .nix?

**Comment by ninelore** [comment]
Also keyd device ids
https://github.com/WeirdTreeThing/cros-keyboard-map/blob/main/cros-keyboard-map.py#L6-L20

**Comment by ChocolateLoverRaj** [comment]
I'll test this once it's completely ready

**Comment by ChocolateLoverRaj** [comment]
> what about the avs dmic config?

@WeirdTreeThing  I think NixOS audio setup is SOF only atm. Idk if there are even any NixOS users who use AVS Chromebooks.

**Comment by ChocolateLoverRaj** [comment]
> Also keyd device ids https://github.com/WeirdTreeThing/cros-keyboard-map/blob/main/cros-keyboard-map.py#L6-L20

@ninelore already done: https://github.com/chrultrabook/docs/commit/889224344007400116bd4bb05c7459f5a6066ed7

**Comment by dj1ch** [comment]
> > what about the avs dmic config?
> 
> @WeirdTreeThing I think NixOS audio setup is SOF only atm. Idk if there are even any NixOS users who use AVS Chromebooks.

there are definitely avs users, iirc at least one having audio issues. @WeirdTreeThing may be referring to [this](https://github.com/thesofproject/avs-topology-xml/tree/main/dmic), although I could be wrong on that one. for that we may just need to clone and use `make` to build it.

**Comment by dj1ch** [comment]
> > Also keyd device ids https://github.com/WeirdTreeThing/cros-keyboard-map/blob/main/cros-keyboard-map.py#L6-L20
> 
> @ninelore already done: [8892243](https://github.com/chrultrabook/docs/commit/889224344007400116bd4bb05c7459f5a6066ed7)

ill add this too

**Comment by WeirdTreeThing** [comment]
> there are definitely avs users, iirc at least one having audio issues. @WeirdTreeThing may be referring to [this repo](https://github.com/thesofproject/avs-topology-xml/tree/main/dmic), although I could be wrong on that one. for that we may just need to clone and use make to build it.

Not what I was talking about. I meant [this](https://github.com/WeirdTreeThing/chromebook-linux-audio/blob/main/conf/avs/51-avs-dmic.lua)

**Comment by ChocolateLoverRaj** [comment]
> Not what I was talking about. I meant [this](https://github.com/WeirdTreeThing/chromebook-linux-audio/blob/main/conf/avs/51-avs-dmic.lua)

@WeirdTreeThing is including that one file enough to get AVS to work on Chromebooks on NixOS?

**Comment by ChocolateLoverRaj** [comment]
> there are definitely avs users, iirc at least one having audio issues

They'll have to test it out then. Are they on GitHub?

**Comment by dj1ch** [comment]
> > there are definitely avs users, iirc at least one having audio issues
> 
> They'll have to test it out then. Are they on GitHub?

i could try and ask them about it, it was on the chrultrabook discord server.

**Comment by WeirdTreeThing** [comment]
>@WeirdTreeThing is including that one file enough to get AVS to work on Chromebooks on NixOS?

No, all it does is make the mic work.

File: src/docs/installing/distros.md [hunk 3]
```diff
@@ -10,13 +10,56 @@ Note that the configurations are applied to `configuration.nix` located in `/etc
...
+    # the rest of your configuration...
+}
+
+```
+
+- First, enable and configure keyd (Example is cros-standard. Adjust as you need!)  
+```nix
+# chrome-device.nix
+{ config, pkgs, lib, ... }:
+
```
Comment (ninelore) [inline src/docs/installing/distros.md:44]: missing surrounding `{}`

File: src/docs/installing/distros.md [hunk 4]
```diff
@@ -77,72 +128,269 @@ services.keyd = {
...
-        mkdir -p $out/share/alsa/ucm2/conf.d
-        cp -r $wttsrc/{hdmi,dmic}-common $wttsrc/GENERATION/* $out/share/alsa/ucm2/conf.d
-
-        runHook postInstall
-        '';
-      });
-    })
+# chrome-device.nix
+{ config, pkgs, lib, ... }:
+
```
Comment (ninelore) [inline src/docs/installing/distros.md:137]: surrounding block missing here aswell

**Review by ninelore — CHANGES_REQUESTED** [review]
Overall weird and possibly broken nix syntax.

**Comment by ChocolateLoverRaj** [comment]
Once the replace dependency thing works with Nix flakes I think it would be nice to have a separate repo with Chromebook config and then NixOS Chromebook users can just add that repo.

File: src/docs/installing/distros.md (hunk 4 above)
Comment (ninelore) [reply src/docs/installing/distros.md:137]: the `services.keyd` block needs to be inside the `in` block

**Comment by ninelore** [comment]
> Once the replace dependency thing works with Nix flakes I think it would be nice to have a separate repo with Chromebook config and then NixOS Chromebook users can just add that repo.

It should be possible

**Comment by ninelore** [comment]
Please leave out the 24.05 syntax for now. rn the page is confusing af

**Comment by ninelore** [comment]
Example config (SOF, cros-standard kb, 24.05)
https://github.com/ninelore/dots.nix/blob/master/nixos/cros-sof-std.nix

**Review by ninelore — CHANGES_REQUESTED** [review]
L177 onwards is still a confusing mess. 

- Merge modprobe part and codeblocks with the wireplumber so we have 1 codeblock for each SOF/AVS and 23.11/24.05 (4 total)
- Use Collapsing sections for those

<details>
<summary>Example</summary>

Foobar

```js
function logSomething(something) {
  console.log('Something', something);
}
```

</details>

```md
<details>
<summary>Example</summary>

Foobar

function logSomething(something) {
  console.log('Something', something);
}

</details>
```

**Comment by ninelore** [comment]
```
Apr 21 14:22:17 9l-lillipup wireplumber[1656]: Old configuration file detected: /etc/wireplumber/main.lua.d/51-increase-headroom.lua
Apr 21 14:22:17 9l-lillipup wireplumber[1656]: Lua configuration files are NOT supported in WirePlumber 0.5. You need to port them to the new format if you w…to use them.
                                               -> See https://pipewire.pages.freedesktop.org/wireplumber/daemon/configuration/migration.html
```

**Comment by ninelore** [comment]
the alsa package override is broken aswell

**Comment by ninelore** [comment]
We have decided not to update the page. Instead it will be removed soon.

**Comment by ChocolateLoverRaj** [comment]
@dj1ch it would be nice if you made your own repo with nix files and instructions for Chromebook audio

**Comment by dj1ch** [comment]
> Reference

perfectly understandable,  thank you anyway.

**Pull request closed without merging**